### ISSUE TITLE
Add WithLayout() to Razor

### ISSUE THREAD

**Issue by daveaglick** [statiqdev/Statiq#186]
We currently have `.WithViewStart()` for the Razor module, but it would be helpful to also have `.WithLayout()` so you could specify a layout for the template without going through the hoop of creating a ViewStart file. Not sure how/if Razor would support this, so may have to hack it in (but see how well the official Razor libs would support - don't want to add this feature only to loose it again when switching to the official libs).

**Comment by deanebarker** [comment]
I think I figured out a graceful way to handle this.  `RazorPage` (and, by inheritance `BaseRazorPage`) has a `Layout` property. Couldn't we just do this...

`WithLayout("Template.cshtml")`

...and set that `Layout` property in the background before we execute?  Clearly, it could be overridden in the cshtml file itself, but that's fine.

**Comment by daveaglick** [comment]
Good call - that's easy enough that it wouldn't be wasted effort if we have to figure out an alternate way with the official Razor library once it comes in (though that won't even be for a while yet).

**Comment by deanebarker** [comment]
I'l get it over the weekend.

**Comment by deanebarker** [comment]
I spent an hour chasing code around this morning.  I cannot figure out, for the life of me, where the base page class is actually instantiated.  I got all the way back into `MvcRazorHost` and that's where I lost the trail.

Even if I can find it, I fear I'm going to be passing that string awfully deep into nested methods to actually get it set. It's not as simple as just changing something in the Module.
